pkg/render/logstorage/esgateway/esgateway.go

```diff
@@ -0,0 +1,352 @@
+package esgateway
+
+import (
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/render"
+	relasticsearch "github.com/tigera/operator/pkg/render/common/elasticsearch"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+)
+
+const (
+	DeploymentName          = "tigera-es-gateway"
+	ServiceAccountName      = "tigera-es-gateway"
+	RoleName                = "tigera-es-gateway"
+	VolumeName              = "tigera-es-gateway-certs"
+	ElasticsearchUserSecret = "tigera-es-gateway-elasticsearch-access"
+	TLSSecret               = "tigera-es-gateway-tls"
+	PortName                = "es-gateway-port"
+	Port                    = 5554
+
+	ElasticsearchHTTPSEndpoint = "https://tigera-secure-internal-es-http.tigera-elasticsearch.svc:9200"
```

Comment:
Adding a new thread for the same line. Although the thread appears to be resolved, nowhere in this PR do I see the creation of a TLS certificate with this SAN. Am I missing anything or will there be TLS issues the way this PR is right now?

Reply:
Never mind, ECK will create those by taking "tigera-secure-internal" and add the suffix "-es-http", right?

Reply:
Yes exactly, everything from service names to certificates are generated by ECK Operator like you said. I've tested this in a fresh cluster and everything works as expected.